### DISCUSSION TITLE
Polkadot, Tron message signing

### DISCUSSION THREAD

**Discussion by AndreiMitrea** [WalletConnect/walletconnect-monorepo#4057] [Developer Support]
Hey. I am trying to implement a flow (that works with web3modal for EVM chains) with Wallet Connect for Polkadot and Tron :

1. Generate a message
2. Prompt the user to sign the message
3. Verify that the correct address was used for the signature.

For Polkadot it works with Nova wallet, but with Trust Wallet I have the following error when the user signs the message:
`{"code":5201,"message":"Unknown method(s) requested"}`

Code is similar with:

```
            params = {
                chainId: 'polkadot:91b171bb158e2d3848fa23a9f1c25182',
                topic: wcSession.topic,
                request: {
                    method: 'polkadot_signMessage',
                    params: {
                        address,
                        message,
                    },
                },
            } as EngineTypes.RequestParams;
            
            ({ signature } = await client.request<{ signature: string }>(params));
```

For Tron, I am able to sign the message with Trust Wallet, but the address returned by `tronWeb.trx.verifyMessageV2`  at the verification step is different to the one used:

```
            const tronWeb = new TronWeb({
                fullHost: 'https://api.trongrid.io',
            });

            const verifiedAddress = await tronWeb.trx.verifyMessageV2(verificationMessage, signature);

```

Please let me know if you have any advice. Thank you!

**Comment by glitch-txs** [comment]
Thanks for reporting, we'll try to communicate this to Trust

**Reply by AndreiMitrea** [reply]
Thank you, @glitch-txs ! Could you please let me know in this discussion if/when the issue is solved?

**Reply by liubin595338764** [reply]
Is there any progress?

**Reply by aminpsycho** [reply]
> Thanks for reporting, we'll try to communicate this to Trust

**Reply by liubin595338764** [reply]
@glitch-txs @aminpsycho  May I ask if you have reproduced this issue? 
Is there any progress in communicating with Trust? 
When is the plan to fix it?

**Reply by glitch-txs** [reply]
Could you share your configuration?

Also could you test it here: https://react-app.walletconnect.com/

**Reply by liubin595338764** [reply]
@glitch-txs It seems like you haven't figured out the problem yet? It is that the signature of connecting TRON network with Trust Wallet cannot be verified.

```js
const message = "This is a message to be signed for Tron";

try {
    const result = await client!.request<{ signature: string }>({
    chainId,
    topic: session!.topic,
    request: {
        method: DEFAULT_TRON_METHODS.TRON_SIGN_MESSAGE,
        params: {
        address,
        message,
        },
    },
    });

    return {
    method: DEFAULT_TRON_METHODS.TRON_SIGN_MESSAGE,
    address,
    valid: true,
    result: result.signature,
    };
```
Your `valid` is fixed to `true`.

Using tronWeb for verification:
```
const verifiedAddress = await tronWeb.trx.verifyMessageV2(message, result.signature);
```
The verified address obtained is not the wallet address

**Reply by glitch-txs** [reply]
Do you know if this issue does NOT happen with other wallet?

**Reply by liubin595338764** [reply]
@glitch-txs Trust is a wallet with a relatively wide user base, I don't have any other Tron wallets.

**Reply by glitch-txs** [reply]
We'll look into this + I've sent this to Trust as well

**Comment by liubin595338764** [comment]
[Emergency] @glitch-txs  I have also encountered the same problem, and I am using `@walletconnect/universal-provider`, `@web3modal/standalone`
```js
const result = await this.provider.client!.request<{
  signature: string
}>({
  topic: this.session?.topic,
  chainId: this.chains[0],
  request: {
    method: 'tron_signMessage',
    params: { message, address },
  },
})
```

```
"@walletconnect/universal-provider": "2.10.0",
"@web3modal/standalone": "2.4.3",
"tronweb": "5.3.1",

trust wallet: 10.6(883)
chrome: 120.0.6099.129
```

I can obtain the signature result normally, but `TronWeb.Trx.verifyMessageV2(msg, signature)` does not return the wallet's address.

**Comment by Sadeghi1017** [comment]
> Thanks for reporting, we'll try to communicate this to Trust

**Comment by liubin595338764** [comment]
@glitch-txs @Sadeghi1017 Is there any progress?

**Comment by dvp82** [comment]
I also encountered a similar problem with signature verification. Is there a solution?
